test/modules/obsagent-client/src/otel_client.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    error::Error,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    time::Duration,
+};
+
+use futures::{Stream, StreamExt};
+use opentelemetry::{
+    global,
+    metrics::{self, ObserverResult},
+    sdk::metrics::{selectors, PushController},
+};
+use opentelemetry_otlp::ExporterConfig;
+use rand::Rng;
+use tracing::error;
+
+use crate::config::Config;
+
+const METER_NAME: &str = "microsoft.com/azureiot-edge";
+
+// Skip first immediate tick from tokio, not needed for async_std.
+fn delayed_interval(duration: Duration) -> impl Stream<Item = tokio::time::Instant> {
+    opentelemetry::util::tokio_interval_stream(duration).skip(1)
+}
+
+fn init_meter(period: f64, otlp_endpoint: String) -> metrics::Result<PushController> {
+    let export_config = ExporterConfig {
+        endpoint: otlp_endpoint,
+        ..ExporterConfig::default()
+    };
+    opentelemetry_otlp::new_metrics_pipeline(tokio::spawn, delayed_interval)
+        .with_export_config(export_config)
+        .with_aggregator_selector(selectors::simple::Selector::Histogram(vec![
+            0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9,
+        ]))
+        .with_period(Duration::from_secs_f64(period))
+        .build()
+}
+
+pub async fn run(config: Arc<Config>) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let _started = init_meter(
+        1.0 / config.otel_config.push_rate,
```

Comment:
A couple things: 
1. This doesn't handle the case where the user provide the rate = 0. Please correct me but I think `unwrap_or` only check if the value exist (if it doesn't, then use the default value). It does not do the check if value is within a valid range.
2. Regarding valid range check, a user can also provide a negative rate as well. We may want to deal with that as well
3. It looks like rust cast the default float as f64, you prob won't need to worry about underflow here. Though you could run into an issue with overflow?

Reply:
I updated the code to throw errors if the rate is negative or zero.

Regarding overflow and underflow, after discussing offline with @yophilav , I decided for simplicity to switch to using periods instead of rates (i.e. update_period and push_period instead of update_rate and push_rate).